### PR TITLE
Make flash syscall interface async

### PR DESCRIPTION
**Pull request by ia0**
This is to support chips that only provide async flash APIs. For chips that only provide blocking flash APIs, the same trick as is used in this PR for the nrf52 chip can be used (using deferred calls).

### CODE REVIEW

File: patches/tock/02-usb.patch [hunk 1]
```diff
@@ -215,8 +215,8 @@ index ebee8704b..6a2f8715b 100644
              kernel::ipc::DRIVER_NUM => f(Some(&self.ipc)),
              _ => f(None),
          }
-@@ -371,6 +388,21 @@ pub unsafe fn reset_handler() {
-         )
+@@ -377,6 +394,21 @@ pub unsafe fn reset_handler() {
+             .expect("no deferred call slot available for nvmc"),
```
Comment (jmichelp) [inline patches/tock/02-usb.patch:219]: Why is USB impacted if we don't change the reference commit for the kernel?

**Review by jmichelp — COMMENTED** [review]
According to the workflows, you forgot to update our own boards (under `boards/`) in addition to those under `third_party/tock/`

File: patches/tock/02-usb.patch (hunk 1 above)
Comment (ia0) [reply patches/tock/02-usb.patch:219]: This is just a context line. The `+` is actually part of the file not part of the diff. It's like for the line numbers that get updated for the patch to be applied perfectly.

**Comment by ia0** [comment]
Yes indeed, I forgot the other boards. I've updated them and also moved the diffs out of the persistent storage patch to the initial fil, now that we don't use the Tock boards at all. This was already the case for the mdk board actually.

File: patches/tock/02-usb.patch (hunk 1 above)
Comment (jmichelp) [reply patches/tock/02-usb.patch:219]: I still read `no deferred call slot available for nvmc` for a patch about USB.
I doubt this was there before ;)

Comment (ia0) [reply patches/tock/02-usb.patch:219]: Yes it wasn't there before this PR. The usb patch is applied after the boards are copied and after the persistent storage patch. So with this PR, when applying the usb patch, this line is actually present.

Comment (jmichelp) [reply patches/tock/02-usb.patch:219]: Gotcha.

**Review by jmichelp — APPROVED** [review]

**Review by kaczmarczyck — APPROVED** [review]
LGTM

If all changes in our own boards are supposed to move out of the patches, I can deal with mine later. Currently, mine is still comprehensive, including all changes to boards and kernel.

**Comment by ia0** [comment]
Indeed:

```
% grep -rl src/main patches/tock 
patches/tock/07-upgrade-partitions.patch
patches/tock/06-firmware-protect.patch
patches/tock/02-usb.patch
```

Do you want to do the move for those files because you're the only one touching at least one of them? (This could be done next time you touch your patch.) Otherwise I can do it when you think the time is right.

Regarding this particular PR, I'll wait to test the dongle before submitting. I only tested the devkit.

**Comment by kaczmarczyck** [comment]
Yes, I thought I'll just do it when I touch the patches!

**Comment by ia0** [comment]
Tested dongle and dev-kit with webauthn (register, login, board reset, login).

**Pull request merged**